MWPW-164491 [Plans] Wide merch card

bozojovicic opened the pull request:
<!-- Before submitting, please review all open PRs. -->

Plans wide and super-wide card with "What's included" component.
Unfortunately wide cards can be properly tested only inside of the card collection, which are not done yet, so I cannot provide any good test page in Milo for this yet.

Resolves: [MWPW-164491](https://jira.corp.adobe.com/browse/MWPW-164491)

**Test URLs:**
- Before: https://main--milo--adobecom.aem.live/?martech=off
- After: https://mwpw164491wide--milo--bozojovicic.aem.live/?martech=off

yesil commented on libs/features/mas/src/merch-card.js:
```diff
@@ -8,6 +8,8 @@ import {
 import './global.css.js';
 import './aem-fragment.js';
+import './merch-mnemonic-list.js';
+import './merch-whats-included.js';
```
I wonder what is the impact of including `merch-whats-included.js in` bytes.
if we render only on clicks, we could also load it on the fly.

npeltier replied:
4 bytes :)?

yesil replied:
not sure, we need to keep in mind that many cards on adobe.com do not need this, especially on page load.

npeltier commented on libs/features/mas/src/utils.js:
```diff
@@ -35,12 +35,16 @@ export function createTag(tag, attributes = {}, content = null, is = null) {
 }
 
+export function matchMobile() {
+  return window.matchMedia('(max-width: 767px)');
```
```suggestion
  return window.matchMedia(MOBILE_LANDSCAPE);
```

npeltier reviewed (approved):
lgtm

afmicka commented:
@bozojovicic 
1. layout for Stock checkbox and Secure transaction in figma is different for wide card without what's included and with it. You have it the same on both. Screenshots to compare studio and figma for wide card. The same is also if i remove the quantity selector from the card. 
- Wide card in studio: 
<img width="670" alt="Screenshot 2025-03-28 at 14 30 43" src="https://github.com/user-attachments/assets/1231ed3b-094a-43c5-8040-d108a0d34c18" />

- Figma: 
<img width="496" alt="Screenshot 2025-03-28 at 14 12 30" src="https://github.com/user-attachments/assets/63c34066-3788-406b-b1d0-40a755770f4c" />
<img width="649" alt="Screenshot 2025-03-28 at 14 11 41" src="https://github.com/user-attachments/assets/8ec85054-21df-4f56-b31b-662f813eacfd" />

2. Callout text seems to be put in the footer for super wide card. If not callout, what is it in figma? In studio is left where it was. 
<img width="855" alt="Screenshot 2025-03-28 at 14 28 11" src="https://github.com/user-attachments/assets/4ba2cfb4-a8b2-4f76-9db6-79faf504bba3" />

3. When previewed in milo (with autoblock), whats included icons are not starting in the new row and are not one bellow another as in studio and figma.  Also for a default (singe) size, those boxes are moved to the right and not aligned anymore for both with and without whats included 
https://mwpw164491wide--milo--bozojovicic.aem.page/drafts/mili/document
<img width="455" alt="Screenshot 2025-03-28 at 14 42 45" src="https://github.com/user-attachments/assets/cfd5c6f8-cb84-4ddf-963b-d384fe1b5f87" />

4. Space is removed before Quantity selector. Was that intentional? I see that space is some figma examples
https://mwpw164491wide--milo--bozojovicic.aem.page/libs/features/mas/docs/plans.html?martech=off
<img width="431" alt="Screenshot 2025-03-28 at 13 56 20" src="https://github.com/user-attachments/assets/d908a8be-cd5b-4b0a-9615-9df708cddb4a" />
<img width="350" alt="Screenshot 2025-03-28 at 13 56 25" src="https://github.com/user-attachments/assets/84983646-04e4-40e1-979f-e6a7b06b52d2" />

bozojovicic commented:
@afmicka 
1. I totally missed this in Figma and Lucy did not mention it in the ticket. I asked her in MWPW-164491 what to do about this case. Waiting ...
2. I agreed with Lucy before to exclude this requirement from this ticket. She created new one for that change MWPW-169832
3. My bad. I fixed it.
4. Lucy asked me if I can reduce some empty space where possible. But this case is really problematic. I reverted my change for that space above the quantity selector. Now it should be as before.

afmicka commented:
@bozojovicic when the card is authored as wide or superwide but viewed as single in milo, the Stock checkbox is not visible
https://mwpw164491wide--milo--bozojovicic.aem.page/drafts/mili/document
<img width="656" alt="Screenshot 2025-04-03 at 10 29 42" src="https://github.com/user-attachments/assets/faad6c07-7e87-48b2-978c-38961f36dc47" />

<img width="396" alt="Screenshot 2025-04-03 at 10 39 06" src="https://github.com/user-attachments/assets/2a84ddc8-e877-4867-9e17-61e838bcbfd8" />
